packages/gatsby-plugin-image/src/components/hooks.ts

```diff
@@ -38,18 +38,37 @@ export function storeImageloaded(cacheKey?: string): void {
 export function hasImageLoaded(cacheKey: string): boolean {
   return imageCache.has(cacheKey)
 }
-
+export type IGatsbyImageDataParent<T = never> = T & {
+  gatsbyImageData: IGatsbyImageData
+}
 export type FileNode = Node & {
-  childImageSharp?: Node & {
-    gatsbyImageData?: IGatsbyImageData
+  childImageSharp?: IGatsbyImageDataParent<Node>
+}
+
+export const isGatsbyImageData = (
+  node: IGatsbyImageData | any
+): node is IGatsbyImageData => Boolean(node?.images?.fallback?.src) // 🦆
+
+export const isGatsbyImageDataParent = <T>(
+  node: IGatsbyImageDataParent<T> | any
+): node is IGatsbyImageDataParent<T> => Boolean(node?.gatsbyImageData)
+
+type ImageDataLike = FileNode | IGatsbyImageDataParent | IGatsbyImageData
+export const getImage = (node: ImageDataLike): IGatsbyImageData | undefined => {
+  if (isGatsbyImageData(node)) {
```

Comment:
Is there a reason for making isGatsbyImageData and isGatsbyImageDataParent? It seems like checking for existence inside this function would be equivalent? I noticed those functions are exported, are they meant for plugins to be able to use?

Reply:
Making them type guard functions allows type narrowing in TS. I exported them so they could be unit tested, but in the end just tested the `getImage` function itself so I could remove the export.

Reply:
Ah ok. Too bad we can't get the benefit without the extra functions. Would make it easier to scan.

Reply:
I think it's easier to read with function name!